service/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java

```diff
@@ -272,14 +310,16 @@ public int delete(long id) {
     giftCertificateRepository
         .getEntity(new GetGiftCertificatesByIdSpecification(id))
         .orElseThrow(
-            () -> new EntityNotFoundException("Requested resource not found (id = " + id + ")"));
+            () ->
```

Comment:
I don't think you need to do this check. Actually, if there is not such certificate, you don't delete anything from DB and return 204. DELETE method is idempotical, so consider to get rid of this check.

Reply:
You're right, thank you. (fixed)